pythonforandroid/graph.py

```diff
@@ -99,17 +132,129 @@ def find_order(graph):
                 bset.discard(result)
 
 
-def get_recipe_order_and_bootstrap(ctx, names, bs=None):
-    recipes_to_load = set(names)
+def obvious_conflict_checker(ctx, name_tuples, blacklist=[]):
+    """ This is a pre-flight check function that will completely ignore
+        recipe order or choosing an actual value in any of the multiple
+        choice tuples/dependencies, and just do a very basic obvious
+        conflict check.
+    """
+    deps_were_added_by = dict()
+    deps = set()
+
+    # Add dependencies for all recipes:
+    to_be_added = [(name_tuple, None) for name_tuple in name_tuples]
+    while len(to_be_added) > 0:
+        current_to_be_added = list(to_be_added)
+        to_be_added = []
+        for (added_tuple, adding_recipe) in current_to_be_added:
+            assert(type(added_tuple) == tuple)
+            if len(added_tuple) > 1:
+                # No obvious commitment in what to add, ignore it.
+                # (Remember this function only catches obvious issues)
+                continue
+
+            name = added_tuple[0]
+            recipe_conflicts = set()
+            recipe_dependencies = []
+            try:
+                # Get recipe to add and who's ultimately adding it:
+                recipe = Recipe.get_recipe(name, ctx)
+                recipe_conflicts = {c.lower() for c in recipe.conflicts}
+                recipe_dependencies = get_dependency_tuple_list_for_recipe(
+                    recipe, blacklist=blacklist
+                )
+            except ValueError:
+                pass
+            adder_first_recipe_name = adding_recipe or name
+
+            # Collect the conflicts:
+            triggered_conflicts = []
+            for dep_tuple_list in deps:
+                # See if the new deps conflict with things added before:
+                if len(set(dep_tuple_list).intersection(
+                       recipe_conflicts)) == len(dep_tuple_list):
+                    triggered_conflicts.append(dep_tuple_list)
+                    continue
+
+                # See if what was added before conflicts with the new deps:
+                if len(dep_tuple_list) > 1:
+                    # Not an obvious commitment to a specific recipe/dep
+                    # to be added, so we won't check.
+                    # (remember this function only catches obvious issues)
+                    continue
```

Comment:
not covered by a test

Reply:
covered in what way? tests that hit this code path are executed, what exactly do you want to be tested? (the very ubiquitous `('python2', 'python3', ..)` tuple hits this code path, as an example)

Reply:
No that code path is NOT executed by a test

Reply:
I added a test case now and verified via print debugging :joy: to see that it's hit. curious, you have to tell me some day how you saw in tox that it's not hit, because I don't actually know how to check that :smile: (I only know how to use independently of tox)

Reply:
You're right, as usual :joy: the problem was that `('python2', 'python3', ...)` are part of the initial set usually and not added in as dependency themselves. I set up a case for that now and at least according to my testing it enters that code path :+1:

Reply:
Nice well done! Well I also put breakpoint or asserts to see it, just like you did.
One day I'll setup code coverage so we can see it from Travis. I do think it matter and I'm very happy you were willing to improve our coverage like you did :beer:

Reply:
@AndreMiras well I am lazy but when I'm poked like that I gotta fix it :joy: so thanks for the poking